### DISCUSSION TITLE
Why you don't use the wslpath command ?

### DISCUSSION THREAD

**Discussion by mveril** [wslutilities/wslpy#36] [Ideas]
For the convert function of `WSLPY` (in the [convert.py](https://github.com/wslutilities/wslpy/blob/3626438b0a0ca5d8a74649a7aff1f9e8dfe46e3f/wslpy/convert.py) file), you use the python code conversion . But since Windows 10 [build 17046](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-17046) (Windows 10 April 2018 Update [[source](https://betawiki.net/wiki/Windows_10_build_17046_(rs_prerelease))]) so a long time ago a command line tool called `wslpath` was added to` WSL` to convert Windows <=> Linux path. One can call it from WSL on any distribution from scratch
A minimal python code snippet
```python
# This is a python minimal sample code to get the Windows path of the current folder from WSL
from subprocess import check_output
print(check_output(['wslpath','-wa',path]).decode().strip())
```
This method ensures good compatibility with all versions that support `wslpath`. This feature was added 3 years ago. So why use a custom conversion instead of calling the wslpath command internally to ensure a perfect result on any computer regardless of the version of Windows or the configuration of the distribution?

**Comment by car-meows** [comment]
Because the wslpy is currently built as the same Windows support as `wslu` which it supports at least Windows 10 Creators Update (15043).  But You do got a good point and will support the `wslpath` for the older builds.

**Reply by mveril** [reply]
Sorry for my late response
I saw that `wslpath` is already used in wslu but I did not see that it is very recent (merged October 4th) wslutilities/wslu#201 so I think wslu does not fully support Windows 10 versions without `wslpath` anymore, (Creators Update and Fall Creators Update). But it's not crazy because these versions are already out of support for Microsoft.
So I think `wslpath` should also be used in `wslpy`.
PS: I tested the same procedure as wslutilities/wslu#201 and there is the same bug in wslpy as described in the previous wslu pull request

**Reply by car-meows** [reply]
> Sorry for my late response
> 
> I saw that `wslpath` is already used in wslu but I did not see that it is very recent (merged October 4th) wslutilities/wslu#201 so I think wslu does not fully support Windows 10 versions without `wslpath` anymore, (Creators Update and Fall Creators Update). But it's not crazy because these versions are already out of support for Microsoft.
> 
> So I think `wslpath` should also be used in `wslpy`.
> 
> PS: I tested the same procedure as wslutilities/wslu#201 and there is the same bug in wslpy as described in the previous wslu pull request
> 
> 

Hmmm, you are right on that. I will move to the `wslpath` method in a future version
